x-pack/auditbeat/cmd/root.go

```diff
@@ -39,12 +49,15 @@ func auditbeatCfg(rawIn *proto.UnitExpectedConfig, agentInfo *client.AgentInfo)
 		return nil, fmt.Errorf("error creating input list from raw expected config: %w", err)
 	}
 
-	// Extract the type field that has "audit/auditd", treat this
-	// as the module config key
-	module := strings.Split(rawIn.Type, "/")[1]
-
-	for iter := range modules {
-		modules[iter]["module"] = module
+	if system.ModuleName == "audit/system" {
+		// running in agentbeat; no need to transform the type field
+	} else {
+		// not running in agentbeat; extract the type field that has
+		// "audit/auditd", treat this as the module config key
+		module := strings.Split(rawIn.Type, "/")[1]
```

Comment:
Without a length check this could panic. IIUC, the goal is to remove the `audit/` prefix, so I think using https://pkg.go.dev/strings#TrimPrefix would be more clear and safer (cannot panic based on user controlled inputs).

Reply:
I was able to just revert this to how it was originally. So this change is no longer in this PR.